app/src/main/java/xyz/whereuat/whereuat/KeyLocDialogFragment.java

```diff
@@ -61,8 +62,20 @@ public void onClick(DialogInterface dialog, int which) {
     @Override
     public void onStart() {
         super.onStart();
-        ((AlertDialog) getDialog()).getButton(AlertDialog.BUTTON_POSITIVE).setTextColor(Color.BLACK);
-        ((AlertDialog) getDialog()).getButton(AlertDialog.BUTTON_NEGATIVE).setTextColor(Color.BLACK);
+        AlertDialog d = (AlertDialog) getDialog();
+        d.getButton(AlertDialog.BUTTON_POSITIVE).setTextColor(Color.BLACK);
+        d.getButton(AlertDialog.BUTTON_NEGATIVE).setTextColor(Color.BLACK);
+
+        EditText e = (EditText) d.findViewById(R.id.key_loc_name_input);
+        e.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (hasFocus) {
+                    AlertDialog d = (AlertDialog) getDialog();
```

Comment:
Are you not able to use the `AlertDialog d` you defined outside of this function?